### PR TITLE
Fix the wrong count of total Pipeline Run number

### LINKED ISSUE #3327: Incorrect count of branch activity in a multi-branch pipeline.
**Describe the Bug**
image

**Preset conditions**
There is a multi-branch pipeline 'pipe-test'.
**Environment**
http://allinone:8080

**How To Reproduce**
1. Go to pipeline 'pipe-test' details page
2. Click on branch master

**Expected behavior**
The number of activities in the master branch is displayed correctly

/kind bug
/area devops
/assign @LinuxSuRen 
/milestone 3.1.0
/priority medium

### PR DESCRIPTION
**Pull request by LinuxSuRen**
**What type of PR is this?**
/kind bug
/area devops
/project devops

**What this PR does / why we need it**:
The count of the total Pipeline Run number is wrong. It makes users feel confused.

**Which issue(s) this PR fixes**:
Fixes #3327

**Special notes for reviewers**:

**Additional documentation, usage docs, etc.**:

### CODE REVIEW

**Comment by LinuxSuRen** [comment]
/cc @kubesphere/sig-devops 

This is the screenshot after it got fixed.

image

File: pkg/simple/client/devops/jenkins/utils.go [hunk 1]
```diff
@@ -91,23 +90,13 @@ func parseJenkinsQuery(query string) (url.Values, error) {
 		if i := strings.Index(key, "="); i >= 0 {
 			key, value = key[:i], key[i+1:]
 		}
-		key, err1 := url.QueryUnescape(key)
-		if err1 != nil {
-			if err == nil {
-				err = err1
+		if key, err = url.QueryUnescape(key); err == nil {
+			if value, err = url.QueryUnescape(value); err == nil {
+				result[key] = append(result[key], value)
```
Comment (shaowenchen) [inline pkg/simple/client/devops/jenkins/utils.go:95]: The previous processing will ignore the error, which causes an incorrect number?

Comment (LinuxSuRen) [reply pkg/simple/client/devops/jenkins/utils.go:95]: The problems come from https://github.com/kubesphere/kubesphere/pull/3335/files#diff-bdbc72911999a583181726d98c9931b3da7070145be9262a56c1b310f6cb9c5dR43

File: pkg/simple/client/devops/jenkins/pure_request.go [hunk 2]
```diff
@@ -34,20 +34,21 @@ func (j *Jenkins) SendPureRequest(path string, httpParameters *devops.HttpParame
 // provider request header to call jenkins api.
 // transfer bearer token to basic token for inner Oauth and Jeknins
 func (j *Jenkins) SendPureRequestWithHeaderResp(path string, httpParameters *devops.HttpParameters) ([]byte, http.Header, error) {
-	Url, err := url.Parse(j.Server + path)
+	apiURL, err := url.Parse(j.Server + path)
 	if err != nil {
 		klog.Error(err)
```
[the remark this answers is not in the document]
Comment (LinuxSuRen) [reply pkg/simple/client/devops/jenkins/pure_request.go]: Done. Please check it from https://github.com/kubesphere/kubesphere/pull/3335/commits/e070f39fa635e66362bb18d3946baac593d9071e

**Comment by shaowenchen** [comment]
There is a conflict that needs to be resolved.

**Pull request merged**